File: backend/services/flow_service.py
from typing import List, Optional
from schemas import ConversationFlow
from data.mock_flows import mock_flows

class FlowService:
    def __init__(self):
        self.flows: List[ConversationFlow] = list(mock_flows)

    def get_all_flows(self) -> List[ConversationFlow]:
        return self.flows

    def get_flow_by_id(self, flow_id: str) -> Optional[ConversationFlow]:
        return next((f for f in self.flows if f.flowId == flow_id or f.id == flow_id), None)

    def detect_flow_from_intent(self, intent: str) -> ConversationFlow:
        lower = intent.lower()
        if 'discount' in lower or 'coupon' in lower:
            return self.get_flow_by_id('discount_inquiry') or self.flows[1]
        if 'order' in lower or 'track' in lower or 'deliver' in lower:
            return self.get_flow_by_id('order_status') or self.flows[2]
        return self.get_flow_by_id('product_search') or self.flows[0]

flow_service = FlowService()

File: backend/src/services/flowService.ts
import { ConversationFlow } from '../types';
import { mockFlows } from '../data/mockFlows';

export class FlowService {
  private flows: ConversationFlow[] = [...mockFlows];

  public getAllFlows(): ConversationFlow[] {
    return this.flows;
  }

  public getFlowById(flowId: string): ConversationFlow | undefined {
    return this.flows.find(f => f.flowId === flowId || f.id === flowId);
  }

  public detectFlowFromIntent(intent: string): ConversationFlow {
    if (intent.includes('discount') || intent.includes('coupon')) {
      return this.getFlowById('discount_inquiry') || this.flows[1];
    }
    if (intent.includes('order') || intent.includes('track') || intent.includes('delivery')) {
      return this.getFlowById('order_status') || this.flows[2];
    }
    if (intent.includes('out_of_stock')) {
      return this.getFlowById('out_of_stock') || this.flows[3];
    }
    return this.getFlowById('product_search') || this.flows[0];
  }
}

export const flowService = new FlowService();
